news/_posts/performance/2020-12-14-apache-vs-nginx.md

```diff
@@ -0,0 +1,161 @@
+* Shops are both configured with 1000 products, 10 suppliers, 2 languages, 5 categories, 1000 commands and a database around 140Mb
+* We ran the same [gatling](https://gatling.io) [crawling scenario](https://github.com/PrestaShop/performance-project), each visitor opening 15 pages per session 
+* The target is to get the most visitors while remaining under 300ms (95th percentile)
+
+Hence the following architecture has been used for our tests:
+
+![TestingArchitecture][TestingArchitecture]
+_The usual suspects_
+{: .text-center }
+
+## FIGHT!
+
+![FIGHT][FIGHT]
+_Garou: Mark of the Wolves belongs to SNK Corporation_
+{: .text-center }
+
+So, we have launched our scenario on each environment and now the results are coming... Let's have a look: 
+
+![VSUsers][VSUsers]
+_Of course, more users the merrier_
+{: .text-center }
+
+Are those the results you were expecting?
+
+## Round 2!
+
+Now, let's see the data we gathered during those tests and check if anything odd appears.
+
+First, the response time and the active users graph built with the Gatling scenario:
+
+![NGINXRespTime][NGINXRespTime]
+_NGINX active users are just below 100, around 94._
+{: .text-center }
+
+![ApacheRespTime][ApacheRespTime]
+_Apache active users are above 100, around 110._
+{: .text-center }
+
+Nothing suprising about the active users, given the result we already had. 
+
+Also, looking at the servers metrics, we see that, for both of them, we have pleny of available memory:
+
+![MemoryUsage][MemoryUsage]
+_Memory usage, not even 10% consumed, we should be fine - excerpt from GCP monitoring_
+{: .text-center }
+
+Same thing for network and disk IO, far from their limits for both instances:
+
+![DiskIO][DiskIO]
+_Nothing impressive here, especially for SSD disks - excerpt from GCP monitoring_
+{: .text-center }
+
+![NetworkTraffic][NetworkTraffic]
+_Network traffric, nothing specific, excerpt from GCP monitoring_
+{: .text-center }
+
+More interesting is the CPU, that is almost completely used for both instances:
+
+![CPUUsage][CPUUsage]
+
+Some may say that looking at the CPU is not that much intesting, but others, such as us, may disagree.
+
+If you remember our previous talk about performance, you may recall that performance tuning is about finding out the current bottleneck of a system. So, if our CPU _is_ the bottleneck, then it is probably not the memory, the disk or even the network.
+
+Also meaning that the CPU can do its part properly.
+
+In our case, we can conclude there is not much constraint on the system preventing it from serving the application as expected - or at least that the constraint are similar enough.
+
+Also, remember that our task here is to use as much ressources as possible and CPU time means it is not waiting for IOs (disk or network), or managing RAM (neither swapping, as it should be seen on disk), and so on.
+
+
+## Finish him!
```

Comment:
Remember that Google will be crawling the HTML tags (this one will be a h2 or h3 after Markdown has been converted to HTML) and show them to everyone in Google Search Results page 😛  you might want to use a title a little more semantic...

Maybe a better title can be used and the gaming reference can be kept as a subtitle?

Reply:
Will work on it ;)